### PR TITLE
chore(eslint): update ESLint and plugins

### LINKED ISSUE #109: An in-range update of @typescript-eslint/eslint-plugin is breaking the build 🚨
## The devDependency [@typescript-eslint/eslint-plugin](https://github.com/typescript-eslint/typescript-eslint) was updated from `2.0.0` to `2.1.0`.

🚨 [View failing branch](https://github.com/visjs/vis-network/compare/master...visjs:greenkeeper%2F%40typescript-eslint%2Feslint-plugin-2.1.0).

This version is **covered** by your **current version range** and after updating it in your project **the build failed**.

@typescript-eslint/eslint-plugin is a devDependency of this project. It **might not break your production code or affect downstream projects**, but probably breaks your build or test tools, which may **prevent deploying or publishing**.

<details>
<summary>Status Details</summary>

- ✅ **ci/circleci: prepare:** Your tests passed on CircleCI! ([Details](https://circleci.com/gh/visjs/vis-network/449?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link)).
- ❌ **ci/circleci: build:** CircleCI is running your tests ([Details](https://circleci.com/gh/visjs/vis-network/452?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link)).
- ❌ **ci/circleci: lint_ts:** Your tests failed on CircleCI ([Details](https://circleci.com/gh/visjs/vis-network/451?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link)).
- ✅ **ci/circleci: test:** Your tests passed on CircleCI! ([Details](https://circleci.com/gh/visjs/vis-network/450?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link)).
</details>

---

<details>
<summary>Release Notes for v2.1.0</summary>

<h1><a href="https://urls.greenkeeper.io/typescript-eslint/typescript-eslint/compare/v2.0.0...v2.1.0">2.1.0</a> (2019-09-02)</h1>
<h3>Bug Fixes</h3>
<ul>
<li><strong>eslint-plugin:</strong> [member-naming] should match constructor args (<a href="https://urls.greenkeeper.io/typescript-eslint/typescript-eslint/issues/771" data-hovercard-type="pull_request" data-hovercard-url="/typescript-eslint/typescript-eslint/pull/771/hovercard">#771</a>) (<a href="https://urls.greenkeeper.io/typescript-eslint/typescript-eslint/commit/b006667">b006667</a>)</li>
<li><strong>eslint-plugin:</strong> [no-inferrable-types] ignore optional props (<a href="https://urls.greenkeeper.io/typescript-eslint/typescript-eslint/issues/918" data-hovercard-type="pull_request" data-hovercard-url="/typescript-eslint/typescript-eslint/pull/918/hovercard">#918</a>) (<a href="https://urls.greenkeeper.io/typescript-eslint/typescript-eslint/commit/a4e625f">a4e625f</a>)</li>
<li><strong>eslint-plugin:</strong> [promise-function-async] Allow async get/set (<a href="https://urls.greenkeeper.io/typescript-eslint/typescript-eslint/issues/820" data-hovercard-type="pull_request" data-hovercard-url="/typescript-eslint/typescript-eslint/pull/820/hovercard">#820</a>) (<a href="https://urls.greenkeeper.io/typescript-eslint/typescript-eslint/commit/cddfdca">cddfdca</a>)</li>
<li><strong>eslint-plugin:</strong> [require-await] Allow concise arrow function bodies (<a href="https://urls.greenkeeper.io/typescript-eslint/typescript-eslint/issues/826" data-hovercard-type="pull_request" data-hovercard-url="/typescript-eslint/typescript-eslint/pull/826/hovercard">#826</a>) (<a href="https://urls.greenkeeper.io/typescript-eslint/typescript-eslint/commit/29fddfd">29fddfd</a>)</li>
<li><strong>eslint-plugin:</strong> [typedef] don't flag destructuring when variables is disabled (<a href="https://urls.greenkeeper.io/typescript-eslint/typescript-eslint/issues/819" data-hovercard-type="pull_request" data-hovercard-url="/typescript-eslint/typescript-eslint/pull/819/hovercard">#819</a>) (<a href="https://urls.greenkeeper.io/typescript-eslint/typescript-eslint/commit/5603473">5603473</a>)</li>
<li><strong>eslint-plugin:</strong> [typedef] handle AssignmentPattern inside TSParameterProperty (<a href="https://urls.greenkeeper.io/typescript-eslint/typescript-eslint/issues/923" data-hovercard-type="pull_request" data-hovercard-url="/typescript-eslint/typescript-eslint/pull/923/hovercard">#923</a>) (<a href="https://urls.greenkeeper.io/typescript-eslint/typescript-eslint/commit/6bd7f2d">6bd7f2d</a>)</li>
<li><strong>eslint-plugin:</strong> [unbound-method] Allow typeof expressions (Fixes <a href="https://urls.greenkeeper.io/typescript-eslint/typescript-eslint/issues/692" data-hovercard-type="issue" data-hovercard-url="/typescript-eslint/typescript-eslint/issues/692/hovercard">#692</a>) (<a href="https://urls.greenkeeper.io/typescript-eslint/typescript-eslint/issues/904" data-hovercard-type="pull_request" data-hovercard-url="/typescript-eslint/typescript-eslint/pull/904/hovercard">#904</a>) (<a href="https://urls.greenkeeper.io/typescript-eslint/typescript-eslint/commit/344bafe">344bafe</a>)</li>
<li><strong>eslint-plugin:</strong> [unbound-method] false positive in equality comparisons (<a href="https://urls.greenkeeper.io/typescript-eslint/typescript-eslint/issues/914" data-hovercard-type="pull_request" data-hovercard-url="/typescript-eslint/typescript-eslint/pull/914/hovercard">#914</a>) (<a href="https://urls.greenkeeper.io/typescript-eslint/typescript-eslint/commit/29a01b8">29a01b8</a>)</li>
<li><strong>eslint-plugin:</strong> [unified-signatures] type comparison and exported nodes (<a href="https://urls.greenkeeper.io/typescript-eslint/typescript-eslint/issues/839" data-hovercard-type="pull_request" data-hovercard-url="/typescript-eslint/typescript-eslint/pull/839/hovercard">#839</a>) (<a href="https://urls.greenkeeper.io/typescript-eslint/typescript-eslint/commit/580eceb">580eceb</a>)</li>
<li><strong>eslint-plugin:</strong> readme typo (<a href="https://urls.greenkeeper.io/typescript-eslint/typescript-eslint/issues/867" data-hovercard-type="pull_request" data-hovercard-url="/typescript-eslint/typescript-eslint/pull/867/hovercard">#867</a>) (<a href="https://urls.greenkeeper.io/typescript-eslint/typescript-eslint/commit/5eb40dc">5eb40dc</a>)</li>
<li><strong>typescript-estree:</strong> improve missing project file error msg (<a href="https://urls.greenkeeper.io/typescript-eslint/typescript-eslint/issues/866" data-hovercard-type="pull_request" data-hovercard-url="/typescript-eslint/typescript-eslint/pull/866/hovercard">#866</a>) (<a href="https://urls.greenkeeper.io/typescript-eslint/typescript-eslint/commit/8f3b0a8">8f3b0a8</a>), closes <a href="https://urls.greenkeeper.io/typescript-eslint/typescript-eslint/issues/853" data-hovercard-type="issue" data-hovercard-url="/typescript-eslint/typescript-eslint/issues/853/hovercard">#853</a></li>
</ul>
<h3>Features</h3>
<ul>
<li>[no-unnecessary-type-assertion] allow <code>as const</code> arrow functions (<a href="https://urls.greenkeeper.io/typescript-eslint/typescript-eslint/issues/876" data-hovercard-type="pull_request" data-hovercard-url="/typescript-eslint/typescript-eslint/pull/876/hovercard">#876</a>) (<a href="https://urls.greenkeeper.io/typescript-eslint/typescript-eslint/commit/14c6f80">14c6f80</a>)</li>
<li><strong>eslint-plugin:</strong> [expl-func-ret-type] make error loc smaller (<a href="https://urls.greenkeeper.io/typescript-eslint/typescript-eslint/issues/919" data-hovercard-type="pull_request" data-hovercard-url="/typescript-eslint/typescript-eslint/pull/919/hovercard">#919</a>) (<a href="https://urls.greenkeeper.io/typescript-eslint/typescript-eslint/commit/65eb993">65eb993</a>)</li>
<li><strong>eslint-plugin:</strong> [no-type-alias] support tuples (<a href="https://urls.greenkeeper.io/typescript-eslint/typescript-eslint/issues/775" data-hovercard-type="pull_request" data-hovercard-url="/typescript-eslint/typescript-eslint/pull/775/hovercard">#775</a>) (<a href="https://urls.greenkeeper.io/typescript-eslint/typescript-eslint/commit/c68e033">c68e033</a>)</li>
<li><strong>eslint-plugin:</strong> add quotes [extension] (<a href="https://urls.greenkeeper.io/typescript-eslint/typescript-eslint/issues/762" data-hovercard-type="pull_request" data-hovercard-url="/typescript-eslint/typescript-eslint/pull/762/hovercard">#762</a>) (<a href="https://urls.greenkeeper.io/typescript-eslint/typescript-eslint/commit/9f82099">9f82099</a>)</li>
<li><strong>typescript-estree:</strong> Accept a glob pattern for <code>options.project</code> (<a href="https://urls.greenkeeper.io/typescript-eslint/typescript-eslint/issues/806" data-hovercard-type="pull_request" data-hovercard-url="/typescript-eslint/typescript-eslint/pull/806/hovercard">#806</a>) (<a href="https://urls.greenkeeper.io/typescript-eslint/typescript-eslint/commit/9e5f21e">9e5f21e</a>)</li>
</ul>
</details>

<details>
<summary>Commits</summary>
<p>The new version differs by 31 commits.</p>
<ul>
<li><a href="https://urls.greenkeeper.io/typescript-eslint/typescript-eslint/commit/6849dc849797289267df17e2425c5098d76ed89f"><code>6849dc8</code></a> <code>chore: publish v2.1.0</code></li>
<li><a href="https://urls.greenkeeper.io/typescript-eslint/typescript-eslint/commit/989c13a359519f40fc8c74cc86c4a017dee94c09"><code>989c13a</code></a> <code>docs(eslint-plugin): [efrt] fix default values in docs</code></li>
<li><a href="https://urls.greenkeeper.io/typescript-eslint/typescript-eslint/commit/65eb99348c8694fc1437f6b601454d2ad0e2834f"><code>65eb993</code></a> <code>feat(eslint-plugin): [expl-func-ret-type] make error loc smaller (#919)</code></li>
<li><a href="https://urls.greenkeeper.io/typescript-eslint/typescript-eslint/commit/6bd7f2d0c0df0361b96509e337e60c77467fde36"><code>6bd7f2d</code></a> <code>fix(eslint-plugin): [typedef] handle AssignmentPattern inside TSParameterProperty (#923)</code></li>
<li><a href="https://urls.greenkeeper.io/typescript-eslint/typescript-eslint/commit/a4e625fc97d05e6bce82f58cd2a877cc35a87820"><code>a4e625f</code></a> <code>fix(eslint-plugin): [no-inferrable-types] ignore optional props (#918)</code></li>
<li><a href="https://urls.greenkeeper.io/typescript-eslint/typescript-eslint/commit/29a01b801063f661c396980810ffbebfd54a8fce"><code>29a01b8</code></a> <code>fix(eslint-plugin): [unbound-method] false positive in equality comparisons (#914)</code></li>
<li><a href="https://urls.greenkeeper.io/typescript-eslint/typescript-eslint/commit/344bafe5c7b17611f496de52f10c0ed556fe81c7"><code>344bafe</code></a> <code>fix(eslint-plugin): [unbound-method] Allow typeof expressions (Fixes #692) (#904)</code></li>
<li><a href="https://urls.greenkeeper.io/typescript-eslint/typescript-eslint/commit/56034736fb3ce6b4d573996a78d408966f67649d"><code>5603473</code></a> <code>fix(eslint-plugin): [typedef] don't flag destructuring when variables is disabled (#819)</code></li>
<li><a href="https://urls.greenkeeper.io/typescript-eslint/typescript-eslint/commit/16136f3ef8a58450e416967d4cb9d1ee0158eae8"><code>16136f3</code></a> <code>docs(eslint-plugin): correct typo in no-unused-vars (#910)</code></li>
<li><a href="https://urls.greenkeeper.io/typescript-eslint/typescript-eslint/commit/5ab13a8d651f02dbf60a41ff2d2ac646f632635b"><code>5ab13a8</code></a> <code>docs(eslint-plugin): no-var-requires: Add example for ES6 modules (#900)</code></li>
<li><a href="https://urls.greenkeeper.io/typescript-eslint/typescript-eslint/commit/cddfdca913b1baa958151e237eac947eaea8f31b"><code>cddfdca</code></a> <code>fix(eslint-plugin): [promise-function-async] Allow async get/set (#820)</code></li>
<li><a href="https://urls.greenkeeper.io/typescript-eslint/typescript-eslint/commit/92e2b31c77bae0c8c541ef2ae5032e1f0a37aae9"><code>92e2b31</code></a> <code>chore(parser): add <code>types</code> field to <code>package.json</code> (#893)</code></li>
<li><a href="https://urls.greenkeeper.io/typescript-eslint/typescript-eslint/commit/29fddfd149599c241d011fb9e1eb2bc830098d4e"><code>29fddfd</code></a> <code>fix(eslint-plugin): [require-await] Allow concise arrow function bodies (#826)</code></li>
<li><a href="https://urls.greenkeeper.io/typescript-eslint/typescript-eslint/commit/6a30de2e00828563d2ddedf5d931f30d07f1a682"><code>6a30de2</code></a> <code>docs: update contributors list</code></li>
<li><a href="https://urls.greenkeeper.io/typescript-eslint/typescript-eslint/commit/2a710233e7b00c811c91f866f5d4a1c258bac250"><code>2a71023</code></a> <code>docs(eslint-plugin): correct typo (#891)</code></li>
</ul>
<p>There are 31 commits in total.</p>
<p>See the <a href="https://urls.greenkeeper.io/typescript-eslint/typescript-eslint/compare/05ba26879dd5a5a0e1159951c8b24dc5e0e5cc4a...6849dc849797289267df17e2425c5098d76ed89f">full diff</a></p>
</details>

<details>
<summary>FAQ and help</summary>

There is a collection of [frequently asked questions](https://greenkeeper.io/faq.html). If those don’t help, you can always [ask the humans behind Greenkeeper](https://github.com/greenkeeperio/greenkeeper/issues/new).
</details>

---

Your [Greenkeeper](https://greenkeeper.io) Bot :palm_tree:

### PR DESCRIPTION
**Pull request by Thomaash**
Closes #109.

### CODE REVIEW

File: lib/network/modules/components/edges/bezier-edge-dynamic.ts [hunk 1]
```diff
@@ -34,7 +34,7 @@ export class BezierEdgeDynamic extends BezierEdgeBase<Point> {
     this._boundFunction = (): void => {
       this.positionBezierNode();
     };
-    this.body.emitter.on("_repositionBezierNodes", this._boundFunction);
+    this._body.emitter.on("_repositionBezierNodes", this._boundFunction);
```
Comment (mojoaxel) [inline lib/network/modules/components/edges/bezier-edge-dynamic.ts]: Out of curiosity: Why the renaming from `body` to `_body`?

Comment (Thomaash) [reply lib/network/modules/components/edges/bezier-edge-dynamic.ts]: We have a rule saying that private and protected members start with underscore and public members can't start with underscore (I added this rule based on how most members were named). Before the update this rule missed this member. Now it is correctly reported.

Comment (mojoaxel) [reply lib/network/modules/components/edges/bezier-edge-dynamic.ts]: I see! (Should have been done in a seperate PR, though)

Comment (Thomaash) [reply lib/network/modules/components/edges/bezier-edge-dynamic.ts]: The rule was already set before, it just caught one more case after this update.

**Pull request merged**